lib/ExecutionEngine/ExecutionEngine.cpp

```diff
@@ -168,15 +166,16 @@ void ExecutionEngine::insertCompiledFunction(
   functionMap[name] = func.get();
   compiledFunctions_[name] = std::move(func);
 
-  std::promise<runtime::ResultCode> addPromise;
+  // TODO: make this a std::promise<llvm::Error>
+  std::promise<bool> addPromise;
```

Comment:
qq: why not make it `std::promise<llvm::Error>` now?

Reply:
Oops good catch thanks! That's what the todo was for. I'll do it now